File: flutterApp/flutter_application_1/lib/models/news.dart
class News {
  final String id;
  final String title;
  final String summary;
  final String content;
  final String category;
  final String coverImage;
  final int viewCount;
  final String createdAt;

  News({
    required this.id,
    required this.title,
    required this.summary,
    required this.content,
    required this.category,
    required this.coverImage,
    required this.viewCount,
    required this.createdAt,
  });

  factory News.fromJson(Map<String, dynamic> json) {
    return News(
      id: json['_id'] ?? '',
      title: json['title'] ?? '',
      summary: json['summary'] ?? '',
      content: json['content'] ?? '',
      category: json['category'] ?? '',
      coverImage: json['coverImage'] ?? '',
      viewCount: json['viewCount'] ?? 0,
      createdAt: json['createdAt'] ?? '',
    );
  }
} 

File: database/models/News.js
// News.js
import mongoose from "mongoose";

const newsSchema = new mongoose.Schema(
  {
    title: {
      type: String,
      required: true,
      trim: true,
    },
    summary: {
      type: String,
      required: true,
      trim: true,
    },
    content: {
      type: String,
      required: true,
    },
    category: {
      type: String,
      required: true,
      enum: ["health", "lifestyle", "medical", "research", "guide"],
    },
    coverImage: {
      type: String,
      required: true,
    },
    author: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      //required: true,
    },
    viewCount: {
      type: Number,
      default: 0,
    },
    GoogleDocsId: {
      type: String,
      default: ''
    },
    viewHistory: {
      type: [{
        date: Date,
        count: Number
      }],
      default: []
    }
  },
  {
    timestamps: true,
  }
);

const News = mongoose.model("News", newsSchema);

export default News;
